wire-ios-data-model/Source/MLS/SafeCoreCrypto.swift

```diff
@@ -42,43 +67,35 @@ public class SafeCoreCrypto: SafeCoreCryptoProtocol {
     private var didInitializeMLS = false
     private let databasePath: String
 
-    public convenience init(coreCryptoConfiguration config: CoreCryptoConfiguration) throws {
+    public convenience init(coreCryptoConfiguration config: CoreCryptoConfiguration) async throws {
         guard let clientID = config.clientIDBytes else {
             throw CoreCryptoSetupFailure.failedToGetClientIDBytes
         }
 
-        let coreCrypto = try CoreCrypto(
-            path: config.path,
-            key: config.key,
-            clientId: clientID,
-            ciphersuites: [defaultCipherSuite.rawValue]
-        )
+        let coreCrypto = try await coreCryptoNew(path: config.path, key: config.key, clientId: clientID, ciphersuites: [CiphersuiteName.default.rawValue], nbKeyPackage: nil)
 
         self.init(coreCrypto: coreCrypto, databasePath: config.path)
         didInitializeMLS = true
     }
 
-    public convenience init(path: String, key: String) throws {
-        let coreCrypto = try CoreCrypto.deferredInit(
-            path: path,
-            key: key,
-            ciphersuites: [defaultCipherSuite.rawValue]
-        )
+    public convenience init(path: String, key: String) async throws {
+        let coreCrypto = try await coreCryptoDeferredInit(path: path, key: key, ciphersuites: [CiphersuiteName.default.rawValue], nbKeyPackage: nil)
 
-        try coreCrypto.setCallbacks(callbacks: CoreCryptoCallbacksImpl())
+        try await coreCrypto.setCallbacks(callbacks: CoreCryptoCallbacksImpl())
 
         self.init(coreCrypto: coreCrypto, databasePath: path)
     }
 
-    public func mlsInit(clientID: String) throws {
+    public func mlsInit(clientID: String) async throws {
         guard !didInitializeMLS else { return }
 
         guard let clientIdBytes = ClientId(from: clientID) else {
             throw CoreCryptoSetupFailure.failedToGetClientIDBytes
         }
-
-        try coreCrypto.mlsInit(clientId: clientIdBytes,
-                               ciphersuites: [defaultCipherSuite.rawValue])
+        // TODO: wait for fix see cyphersuite cyphersuiteName
+        try await coreCrypto.mlsInit(clientId: clientIdBytes,
+                                     ciphersuites: [CiphersuiteName.default.rawValue],
```

Comment:
We could avoid the problem using different configurations for cipherSuites at different places by fetching it from a class variable instead of setting it here

Reply:
Hmm, yeah let review this. maybe this whole thing actually move to `CoreCryptoProvider`.